lib/file-utilities.js

```diff
@@ -102,64 +104,50 @@ function getFiles(config, dir, callback) {
 
 function streamFiles(config, logfiles, callback) {
   logfiles.forEach((file) => {
-    let tail
-
-    if (os.platform() !== 'win32' && config.TAIL_MODE === 'unix') {
-      debug('tailing: ' + file)
-      tail = spawn('tail', ['-n0', '-F', file])
-      tail.stdout.on('data', (data) => {
-        const lines = data.toString().trim().split('\n')
-        for (const line of lines) {
-          if (config.exclude_regex && config.exclude_regex.test(line)) continue
-          client.log.agentLog({
-            line
-          , f: file
-          })
-        }
-      })
+    const startPos = 0
+    const tail = new TailFile(file, {
+      encoding: 'utf8'
+    , startPos
+    })
+    debug(`tailing: ${file}`)
+    tail.on('error', (error) => {
+      log(`tail error: ${file}: ${error}`, 'error')
+    })
+
+    tail.on('end', (error) => {
+      if (error) {
+        log(
+          `file does not exist, stopped tailing: ${file} after ${tail.timeout}ms`, 'warn'
+        )
+        files = files.filter((element) => { return element !== file })
+      }
+    })
 
-      tail.stderr.on('data', (error) => {
-        log(`tail error: ${file}: ${error.toString().trim()}`, 'error')
-      })
+    tail.on('rename', () => {
+      log(`log rotated: ${file} by rename`)
+    })
 
-      tails.push(tail)
+    tail.on('truncate', () => {
+      log(`log rotated: ${file} by truncation`)
+    })
 
-    } else {
-      debug(`tailing: ${file}`)
-      tail = TailReadStream.tail(file, config)
-      tail.pipe(new Splitter())
+    try {
+      tail.start()
+      tail.pipe(split2())
         .on('data', (line) => {
           if (config.exclude_regex && config.exclude_regex.test(line)) return
           client.log.agentLog({
             line
           , f: file
           })
         })
-
-      tail.on('error', (error) => {
-        log(`tail error: ${file}: ${error}`, 'error')
-      })
-
-      tail.on('end', (error) => {
-        if (error) {
-          log(`file does not exist, stopped tailing: ${file} after ${tail.timeout}ms`, 'warn')
-          files = files.filter(element => element !== file)
-        }
-      })
-
-      tail.on('rename', () => {
-        log(`log rotated: ${file} by rename`)
-      })
-
-      tail.on('truncate', () => {
-        log(`log rotated: ${file} by truncation`)
-      })
-
       tailStreams.push(tail)
+    } catch (err) {
```

Comment:
This doesn't actually work the way you intend because `.start()` returns a promise which needs to be `await`ed before an error would throw.  You can't do that here because we're not in an `async` function.  . Data also won't flow until you pipe, so you can set up your data listener first.  The correct way is to use thenables all throughout:

```js
tail.on('data', ...)
tail
  .start()
  .then(() => {
    if (callback) callback()
  })
  .catch((err) => {
     log.error(err, 'Tail could not start. Does the file exist?')
  })
```